### PR TITLE
Simplify HuggingFace Integration

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes
* Fixed mypy issues with transformers. (using inout from huggingface Issue #13390)
* Removed mlflow to make the example a bit more lightweight.
* Fixed import of copy_dir

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: examples/huggingface/README.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
-# Implementation of NLP algorithms using Hugging Face & ZenML
+# 🤗 Implementation of NLP algorithms using Hugging Face & Zenml
```
Comment (htahir1) [inline examples/huggingface/README.md:1]: Should this not be `ZenML` actually?

File: pyproject.toml [hunk 2]
```diff
@@ -177,7 +177,10 @@ show_error_codes = true
 exclude = "airflow/"
 
 [[tool.mypy.overrides]]
-module = "airflow.*"
+module = [
```
Comment (htahir1) [inline pyproject.toml:180]: I think this and the mypy overrides below should be merged

Comment (htahir1) [reply pyproject.toml:180]: @schustmi Do you know why exactly we have two different `[[tool.mypy.overrides]]`?

**Review by htahir1 — CHANGES_REQUESTED** [review]
Almost LGTM! Just a few comments

File: pyproject.toml (hunk 2 above)
Comment (AlexejPenner) [reply pyproject.toml:180]: Was wondering the same thing, one of them explicitly has a  `follow_imports = "skip"` and the other one doesn't so with the release at 13 coming up I felt like it was save to leave this question for next time.

File: examples/huggingface/README.md (hunk 1 above)
Comment (AlexejPenner) [reply examples/huggingface/README.md:1]: yeah definitely

File: pyproject.toml (hunk 2 above)
Comment (AlexejPenner) [reply pyproject.toml:180]: https://mypy.readthedocs.io/en/stable/config_file.html - if you scroll down here it seems desired to have separate [[tool.mypy.overrides]]
instances that descrive different configurations for the associated modules

Comment (schustmi) [reply pyproject.toml:180]: The override section here is just a temporary fix for a specific version of airflow which should get removed once we update the dependency version. The second `follow_imports = "skip"` is intended for any dependency that is not airflow and should not be imported by mypy. However there are very few reasons to do this and we should use it carefully, usually putting it in our third section with `ignore_missing_imports=true` should be enough and is the preferred way if some type annotations are missing.

Comment (AlexejPenner) [reply pyproject.toml:180]: added to the second as we need to skip the follow_imports for now as recommended by huggingface

**Review by htahir1 — APPROVED** [review]
LGTM!

**Pull request merged**